src/vm/types/signatures.rs

```diff
@@ -1054,6 +1045,12 @@ impl TypeSignature {
         }
         Ok(trait_signature)
     }
+
+    pub fn from_string(val: &str, version: ClarityVersion) -> Self {
+        use vm::ast::parse;
+        let expr = &parse(&QualifiedContractIdentifier::transient(), val, version).unwrap()[0];
+        TypeSignature::parse_type_repr(expr, &mut ()).unwrap()
+    }
```

Comment:
Can you mark this `#[cfg(test)]`? The previous `From<&str>` implementation was mostly just used as a convenience for constructing test assertions, and I wouldn't want to see this getting used in production.